skrub/tests/test_sentence_encoder.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+import pytest
+from numpy.testing import assert_array_equal
+from sklearn.base import clone
+
+from skrub import SentenceEncoder
+from skrub._on_each_column import RejectColumn
+from skrub._sentence_encoder import ModelNotFound
+
+pytest.importorskip("sentence_transformers")
+
+
+@pytest.fixture
+def encoder():
+    """Create a fixture such that we set the common parameters of the encoder.
+
+    Here, we have two constraints:
+
+    - We want a really small model to not blow up the memory of the GitHub Actions
+      workers.
+    - We want to force the use of the CPU. GitHub Actions workers on macOS ARM64 will
+      detect the MPS backend, but due to limitations, no memory can be allocated.
+      See https://github.com/actions/runner-images/issues/9918 for more details.
+    """
+    return SentenceEncoder(
+        model_name="sentence-transformers/paraphrase-albert-small-v2",
+        device="cpu",
+    )
+
+
+def test_missing_import_error(encoder):
+    try:
+        import sentence_transformers  # noqa
+    except ImportError:
+        pass
+    else:
+        return
+
+    st = clone(encoder)
+    x = pd.Series(["oh no"])
+    with pytest.raises(ImportError, match="Missing optional dependency"):
+        st.fit(x)
+
+
+def test_sentence_encoder(df_module, encoder):
+    X = df_module.make_column("", ["hello sir", "hola que tal"])
+    encoder = clone(encoder).set_params(n_components=2)
+    X_out = encoder.fit_transform(X)
+    assert X_out.shape == (2, 2)
```

Comment:
it's a detail but .shape only happens to work because pandas and polars use the same name for it, so if we want to be really nitpicky we would write sbd.shape(X_out)

Reply:
right, I thought that for simple case like this one sbd wasn't necessary. Ok to use it though

Reply:
maybe one day we will add ibis whose tables don't have a shape :)